src/Docker/Runner.php

```diff
@@ -398,17 +425,37 @@ private function waitForStorageJobs(array $outputs)
      * @param string $configVersion
      * @param string $mode
      * @param InputTableStateList $inputTableStateList
+     * @param InputFileStateList $inputFileStateList
      * @return Output
      * @throws ClientException
      */
-    private function runComponent($jobId, $configId, $rowId, Component $component, UsageFileInterface $usageFile, DataLoaderInterface $dataLoader, WorkingDirectory $workingDirectory, StateFile $stateFile, ImageCreator $imageCreator, ConfigFile $configFile, OutputFilterInterface $outputFilter, $configVersion, $mode, InputTableStateList $inputTableStateList)
-    {
+    private function runComponent(
+        $jobId,
+        $configId,
+        $rowId,
+        Component $component,
+        UsageFileInterface $usageFile,
+        DataLoaderInterface $dataLoader,
+        WorkingDirectory $workingDirectory,
+        StateFile $stateFile,
+        ImageCreator $imageCreator,
+        ConfigFile $configFile,
+        OutputFilterInterface $outputFilter,
+        $configVersion,
+        $mode,
+        InputTableStateList $inputTableStateList,
+        InputFileStateList $inputFileStateList
+    ) {
         // initialize
         $workingDirectory->createWorkingDir();
-        $resultInputTablesState = $dataLoader->loadInputData($inputTableStateList);
+        list($resultInputTablesState, $resultInputFilesState) = $dataLoader->loadInputData(
+            $inputTableStateList,
+            $inputFileStateList
+        );
 
         $output = $this->runImages($jobId, $configId, $rowId, $component, $usageFile, $workingDirectory, $imageCreator, $configFile, $stateFile, $outputFilter, $dataLoader, $configVersion, $mode);
-        $output->setInputTableStateList($resultInputTablesState);
+        $output->setInputTableStateList($resultInputTablesState ? $resultInputTablesState : new InputTableStateList([]));
```

Comment:
what's the teranary here for?

Reply:
the files needed it, so I put it here too

Reply:
this will be solved by the new object too.

Reply:
I don't see a way how loadInputData could return anything else then InputTableStateList and InputFileStateList

Reply:
me neither!  but for some reason the synapse runner test returned `null` for `InputFileStateList`

Reply:
so there's a bug then i suppose, but i don't really see how it would be happening

Reply:
@odinuv here it is if you're still interested https://dev.azure.com/keboola-dev/docker-runner/_build/results?buildId=9261&view=logs&j=00172958-dc2f-5a07-7f65-b85e47f36adc&t=d87b4090-a97e-58d5-e3ab-66384d653296&l=2036